tests/pipelines/test_pipelines_token_classification.py

```diff
@@ -580,6 +594,41 @@ def test_gather_pre_entities(self):
             ],
         )
 
+    @require_torch
+    def test_word_heuristic_leading_space(self):
+        model_name = "microsoft/deberta-base"
+        tokenizer = AutoTokenizer.from_pretrained(model_name, use_fast=True)
+        token_classifier = pipeline(task="ner", model=model_name, tokenizer=tokenizer, framework="pt")
+
+        sentence = "I play the harmonica"
+
+        tokens = tokenizer(
+            sentence,
+            return_attention_mask=False,
+            return_tensors="pt",
+            return_special_tokens_mask=True,
+            return_offsets_mapping=True,
+        )
+        offset_mapping = tokens.pop("offset_mapping").cpu().numpy()[0]
+        special_tokens_mask = tokens.pop("special_tokens_mask").cpu().numpy()[0]
+        input_ids = tokens["input_ids"].numpy()[0]
+        scores = np.array([[1, 0] for _ in input_ids])  # values irrelevant for heuristic
+
+        pre_entities = token_classifier.gather_pre_entities(
+            sentence,
+            input_ids,
+            scores,
+            offset_mapping,
+            special_tokens_mask,
+            aggregation_strategy=AggregationStrategy.FIRST,
+        )
+
+        # check is_subword values for each token - "harmonica" is two tokens
```

Comment:
Great small tests.

I seem removing your modification didn't lead to this test failing. I may have gotten it wrong, I will try again tomorrow.

I think we should add something like

```python
 self.assertEqual(
            [entity["word"] for entity in pre_entities],
            # play is NOT a subword of `"I" `
            ["I", " play", " the" " h", "armonica"],        )
```
This is mostfly for the test reader so he can reason about the `is_subword` values without having to dive into the code.

Reply:
Making that clearer is a good idea, will add that

Reply:
And I believe both tests failed before the change, but will double check as well

Reply:
Please ignore.

Reply:
Everything works. I probably made a mistake like forgetting to save or something earlier.